app/src/main/java/com/github/epfl/meili/database/Database.kt

```diff
@@ -1,11 +1,13 @@
 package com.github.epfl.meili.database
 
 import java.util.*
+import kotlin.collections.HashSet
 
-abstract class Database<T>(private val path: String): Observable() {
-    abstract var values: Map<String, T>
 
-    abstract fun addElement(uid: String?, element: T?)
+abstract class Database<T>(private val key: String): Observable() {
```

Comment:
Personally, I believe that `path` as variable name  was clearer than `key`. What do you think? Did you have a strong reason to change it?

Reply:
Okay, you're right, since you might want to have more complex paths than simple document keys.

Reply:
Great! Thank you!